design-docs/008-command-line-interface.md

```diff
@@ -0,0 +1,143 @@
+# Icechunk Command Line Interface
+
+This document outlines the design of the Icechunk command line interface.
+
+## Functionality
+
+Here is a list of tasks a user might want to do with Icechunk:
+
+- List my repositories
+- List a history of a repo
+- List branches in a repo
+- List tags in a repo
+- Create a new repository
+- Check configuration
+- Diff between two commits
+- Invoke administrative tasks (garbage collection, compaction, etc)
```

Comment:
This is not yet reflected in the current `Interface` section. Detailed API is out of scope, but it might be useful to think about this to see if the suggested command structure works.